activesupport/test/core_ext/module/attribute_accessor_per_thread_test.rb

```diff
@@ -4,21 +4,37 @@
 require "active_support/core_ext/module/attribute_accessors_per_thread"
 
 class ModuleAttributeAccessorPerThreadTest < ActiveSupport::TestCase
+  class MyClass
+    thread_mattr_accessor :foo
+    thread_mattr_accessor :bar,  instance_writer: false
+    thread_mattr_reader   :shaq, instance_reader: false
+    thread_mattr_accessor :camp, instance_accessor: false
+  end
+
+  class SubMyClass < MyClass
+  end
+
   def setup
-    @class = Class.new do
-      thread_mattr_accessor :foo
-      thread_mattr_accessor :bar,  instance_writer: false
-      thread_mattr_reader   :shaq, instance_reader: false
-      thread_mattr_accessor :camp, instance_accessor: false
+    @class = MyClass
+    @subclass = SubMyClass
+    @object = @class.new
+  end
 
-      def self.name; "MyClass" end
-    end
+  def test_can_initialize_with_default_value
+    Thread.new do
+      @class.thread_mattr_accessor :baz, default: "default_value"
+      assert_equal "default_value", @class.baz
```

Comment:
`"\n` added. 

> Should we add the same assertion after the join to see that it was actually set in the other thread?

🤔 good point, the implementation I have here sets the default per thread, so different threads can initialize with different defaults, but I think your suggestion here makes sense, we probably want to share defaults across all threads.

Reply:
Sorry, I meant that "it wasn't actually set" because we don't want the values to cross over. Saw you updated the test to check that there was no cross over, sweet!